datalad/metadata/metadata.py

```diff
@@ -172,11 +186,11 @@ def _get_containingds_from_agginfo(info, rpath):
     return dspath
 
 
-def query_aggregated_metadata(reporton, ds, aps, recursive=False,
-                              **kwargs):
+def legacy_query_aggregated_metadata(reporton, ds, aps, recursive=False,
```

Comment:
correspondingly `gen2_query_...`

Reply:
The `query_aggregated_metadata` method from master would actually be compatible with at least `gen2` and `gen3` (if I am not mistaken, @mih could clarify that).

I would therefore still rename it to `legacy_query_aggregated_metadata` because it covers two generations (due to the the shared storage model),